File: TooksCms.ServiceLayer/Gadgets/Gadget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using TooksCms.Core.Bases;
using TooksCms.Core.Interfaces;

namespace TooksCms.ServiceLayer.Gadgets
{
    public class Gadget : InterfacingBase, IGadget
    {
        public Gadget() { }

        public Gadget(IGadget data) : 
            base(data, typeof(IGadget))
        {
        
        }

        #region IGadget Members

        public int GadgetId { get; set; }

        public Guid GadgetUid { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string View { get; set; }

        public int SiteId { get; set; }

        public int DefaultColumn { get; set; }

        #endregion
    }
}

File: TooksCms.Web/App/models/gadget.js
define('models/gadget', function () {
    "use strict";

    var Gadget = function (data) {
        this.gadgetId = data.gadgetId;
        this.name = data.name;
        this.description = data.description;
        this.view = data.view;
        this.defaultColumn = data.defaultColumn;
        this.roleName = data.roleName;
        this.areaType = data.areaType;
    }

    return {
        gadget: Gadget
    };
});